mobile/lib/features/wallet/share_invoice_screen.dart

```diff
@@ -141,4 +164,29 @@ class ShareInvoiceScreen extends StatelessWidget {
       )),
     );
   }
+
+// Pay the generated invoice with 10101 faucet
+// XXX: This is in not in Rust as this is not production-related
+  Future<void> payInvoiceWithFaucet(String invoice) async {
+    // Default to the faucet on the 10101 server, but allow to override it
+    // locally if needed for dev testing
+    // It's not populated in Config struct, as it's not used in production
+    String faucet =
+        const String.fromEnvironment("REGTEST_FAUCET", defaultValue: "http://35.189.57.114:8080");
+
+    final data = {'payment_request': invoice};
+    final encodedData = json.encode(data);
+
+    final response = await http.post(
+      Uri.parse('$faucet/lnd/v1/channels/transactions'),
+      headers: <String, String>{'Content-Type': 'application/json'},
+      body: encodedData,
+    );
+
+    if (response.statusCode == 200) {
+      FLog.info(text: response.body);
+    } else {
+      FLog.error(text: "error: ${response.statusCode} body: ${response.body}");
```

Comment:
:wrench: Since this is something that we know often fails, I think it would be good to give feedback to the user beyond printing an error log.

In this PR that was not merged I [used a snack bar to display the error message](https://github.com/get10101/10101/pull/533/files#diff-e8ba3fa73b3c5a44a61895c9196f7d935a42e33ac87bad9973665d64efdadac7R79-R80) and it was really simple.

It might also be a good idea to not navigate away from the share-invoice screen if this fails.